plugins/inputs/mavlink/mavlink.go

```diff
@@ -0,0 +1,123 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mavlink
+
+import (
+	_ "embed"
+	"fmt"
+
+	"github.com/chrisdalke/gomavlib/v3"
+	"github.com/chrisdalke/gomavlib/v3/pkg/dialects/ardupilotmega"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Plugin state
+type Mavlink struct {
+	// Config param
+	FcuURL                 string   `toml:"fcu_url"`
+	SystemID               uint8    `toml:"system_id"`
+	MessageFilter          []string `toml:"message_filter"`
+	StreamRequestEnable    bool     `toml:"stream_request_enable"`
+	StreamRequestFrequency int      `toml:"stream_request_frequency"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	// Internal state
+	connection     *gomavlib.Node
+	endpointConfig []gomavlib.EndpointConf
+	terminated     bool
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Mavlink) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Mavlink) Init() error {
+	// Parse out the Mavlink endpoint.
+	endpointConfig, err := ParseMavlinkEndpointConfig(s)
+	if err != nil {
+		return fmt.Errorf("%s", err.Error())
```

Comment:
Simply return the error? Why wrapping the message in a new string to be added as error?

Reply:
Fixed